packages/blade/src/components/Chip/Chip.stories.tsx

```diff
@@ -0,0 +1,291 @@
+  isDisabled: true,
+};
+
+const ChipWithIconTemplate: ComponentStory<typeof ChipComponent> = ({ children, ...args }) => {
+  return (
+    <Box>
+      <Text marginBottom="spacing.3" marginTop="spacing.3">
+        What other capabilities are you looking for?
+      </Text>
+
+      <ChipGroupComponent
+        accessibilityLabel="Choose one business type from the options below"
+        defaultValue="payment-links"
+        {...args}
+      >
+        <ChipComponent value="payment-links" icon={PaymentLinksIcon} {...args}>
+          Automated Payment Links
+        </ChipComponent>
+        <ChipComponent value="wallet" icon={SmartphoneIcon} {...args}>
+          Wallet on My App
+        </ChipComponent>
+        <ChipComponent value="offers" icon={TagIcon} {...args}>
+          Offer discounts, Pay Later & EMI options
+        </ChipComponent>
+      </ChipGroupComponent>
+    </Box>
+  );
+};
+
+export const ChipWithIcon = ChipWithIconTemplate.bind({});
+ChipWithIcon.storyName = 'With Icon';
+
+const ChipIntentsTemplate: ComponentStory<typeof ChipComponent> = (args) => {
+  return (
+    <Box display="flex" flexDirection="column">
+      <Text size="large" weight="bold" marginBottom="spacing.3">
+        Is the result helpful?
+      </Text>
+
+      <ChipGroupComponent
+        defaultValue="yes"
+        accessibilityLabel="Is the result helpful? Please select either yer or no"
+      >
+        <ChipComponent intent="positive" value="yes" icon={ThumbsUpIcon} {...args}>
+          Yes
+        </ChipComponent>
+        <ChipComponent intent="negative" value="no" icon={ThumbsDownIcon} {...args}>
+          No
+        </ChipComponent>
+      </ChipGroupComponent>
+    </Box>
+  );
+};
+
+export const ChipWithIntent = ChipIntentsTemplate.bind({});
+ChipWithIntent.storyName = 'With Intent';
+
+const TextTransformationTemplate: ComponentStory<typeof ChipComponent> = ({
+  children,
+  ...args
+}) => {
+  const chipValues = ['Proprietorship', 'Public', 'Small Business'];
+  return (
+    <Box>
+      <Text marginBottom="spacing.3" marginTop="spacing.3">
+        Select Business type:
+      </Text>
+
+      <ChipGroupComponent
+        accessibilityLabel="Choose one business type from the options below"
+        defaultValue="Proprietorship"
+      >
+        {chipValues.map((chipValue: string) => (
+          <ChipComponent key={chipValue} value={chipValue} {...args}>
+            {chipValue.toUpperCase()}
+          </ChipComponent>
+        ))}
+      </ChipGroupComponent>
+      <Text marginTop="spacing.3">
+        The text within the Chip can be transformed to uppercase by passing{' '}
+        <Code size="medium"> value.toUpperCase() </Code> as the children.
+      </Text>
+    </Box>
+  );
+};
+
+export const TextTransformationUppercase = TextTransformationTemplate.bind({});
+TextTransformationUppercase.storyName = 'Text Transformation (Uppercase)';
+
+export const chipRef: ComponentStory<typeof ChipComponent> = (args) => {
```

Comment:
chip ref doesn't makes sense for `Chip` component so just keep this only in `ChipGroup`